packages/map/src/map/hooks.jsx

```diff
@@ -1,65 +1,69 @@
-import { useEffect, useState, useRef, useContext } from 'react';
-import ReactMapboxGl from 'react-mapbox-gl';
+import { useEffect, useState, useContext, useMemo } from 'react';
+import { Map, NavigationControl } from 'mapbox-gl';
+
+import { getMapOptions, getBoundingBox, mergeChildrenBounds } from './utils';
 import { getMedia, media } from '../utils';
 import MapContext from './context';
 
 
-export const useLocked = (locked, lockedMobile) => {
-  const [isLocked, setLocked] = useState(locked);
+export const useMapContext = () => useContext(MapContext);
+
+export const useMapInstance = (nodeRef, options) => {
+  const [mapInstance, setMap] = useState(false);
+  const { noAttribution, locked, lockedMobile, onLoad } = options;
 
   useEffect(() => {
-    const isMobile = !getMedia(global, media.mediaM);
-    setLocked(isMobile ? lockedMobile : locked);
-  }, [locked, lockedMobile]);
+    const mapOptions = getMapOptions({
+      ...options,
+      node: nodeRef.current,
+      isMobile: !getMedia(global, media.mediaM),
+    });
 
-  return isLocked;
-};
+    const map = new Map(mapOptions);
 
-export const useMapboxComponent = (isLocked, noAttribution) => {
-  const [, setState] = useState();
-  const ref = useRef(null);
+    const handleLoad = () => {
+      if (onLoad) onLoad();
```

Comment:
`invoke`?

Reply:
don't want to add dep only for one helper